### PR TITLE
feat(macos): Expose tao event `Event::Reopen`, closes #3084

### LINKED ISSUE #3084: [feat][macOS] Provide a hook to capture the dock events / activate event
**Is your feature request related to a problem? Please describe.**
I use api.prevent_close() and window.hide() to prevent the app to exit. And then I want to show the window again by left-clicking the icon on the dock, but it does not work. The question is how to capture the left-click events?

**Describe the solution you'd like**
Provide a hook to capture theses events,  like this:
```
  app.run(|app_handle, event| match event {
    DockEvent::LeftClick => {}
    _ => {}
)
```

**Describe alternatives you've considered**
None.

**Additional context**
image

**Cargo tauri info**
Operating System - Mac OS, version 11.6.1 X64

Node.js environment
  Node.js - 14.18.1
  @tauri-apps/cli - 1.0.0-beta.10
  @tauri-apps/api - 1.0.0-beta.8

Global packages
  npm - 6.14.15
  yarn - 1.22.17

Rust environment
  rustc - 1.56.1
  cargo - 1.56.0

App directory structure
/bin
/node_modules
/webpack
/public
/src-web
/src-tauri
/change-logs
/build
/.git
/src

App
  tauri.rs - 1.0.0-beta.8
  build-type - bundle
  CSP - default-src blob: data: filesystem: ws: wss: http: https: tauri: 'unsafe-eval' 'unsafe-inline' 'self' img-src: 'self'
  distDir - ../build
  devPath - http://localhost:9091
  framework - React
  bundler - Webpack

### PR DESCRIPTION
**Pull request by BillGoldenWater**
…for handle click on dock icon on macOS.

### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### Other information
in macOS the app usually doesn't exit when all window closed, and will use click on dock icon for reopen window.

---

depends on tauri-apps/tao#517, tauri-apps/tao#913

### CODE REVIEW

**Comment by pronebird** [comment]
@amrbashir Hi! Could you please prioritize this if possible. I am really yearning for this to finally become a part of tauri! Thank you! 😄

**Comment by pronebird** [comment]
Linked PR: https://github.com/tauri-apps/tao/pull/517

File: .changes/event-reopen.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'tauri': 'minor:feat'
+'tauri-runtime': 'minor:feat'
+'tauri-runtime-wry': 'minor:feat'
+---
+
+Expose tao event `Event::Reopen` for handle click on dock icon on macOS.
```
Comment (amrbashir) [inline .changes/event-reopen.md]:
```suggestion
Add `RunEvent::Reopen` for handle click on dock icon on macOS.
```

**Comment by amrbashir** [comment]
@BillGoldenWater a new tao version has been released, you can update it in your PR

**Review by amrbashir — APPROVED** [review]

**Comment by amrbashir** [comment]
Thanks @BillGoldenWater

**Pull request merged**